network_transactions.py

```diff
@@ -52,86 +53,135 @@ def visualize_network(graph, fraud_edges=None):
         x = (pos[u][0] + pos[v][0]) / 2  # Midpoint x-coordinate
-        y = (pos[u][1] + pos[v][1]) / 2 + edge_label_pos_offset  # Midpoint y-coordinate + offset
-        plt.text(x, y, label, color='red', ha='center', va='bottom')  # Adjust the vertical alignment
+        y = (
+            pos[u][1] + pos[v][1]
+        ) / 2 + edge_label_pos_offset  # Midpoint y-coordinate + offset
+        plt.text(
+            x, y, label, color="red", ha="center", va="bottom"
+        )  # Adjust the vertical alignment
     # remove axis
-    plt.axis('off')
+    plt.axis("off")
     # render the graph
     plt.show()
 
 
 #
 # #####
 # the fraud_detection function
-#   input : graoh - this is the networkx object that contains all
+#   input : graph - this is the networkx object that contains all
 #                   the nodes and edges for from our transactions.cvs file
-#           threshold - any transaction that exceeds this ammount will be flagged as
+#           threshold - any transaction that exceeds this amount will be flagged as
 #                    potential fraud. The default is $500.00
 #   output : the fraud_edges , this will be used by the visualization function
 def fraud_detection(graph, threshold=500.00):
     #   find the list of edges that represent transactions greater
     #       than the threshold
-    fraud_edges = [(u, v) for u, v in graph.edges() if graph[u][v]['amount'] > threshold]
+    fraud_edges = [
+        (u, v) for u, v in graph.edges() if graph[u][v]["amount"] > threshold
+    ]
     #   print the list of fraud edges:   the source person, destination person,
     #       and the transaction amount
     for sender, receiver in fraud_edges:
-        amount = graph[sender][receiver]['amount']
-        print(f"Potential Fraud Detected (Amount Exceeds $500.00): {sender} -> {receiver}, Amount: ${amount:.2f}")
+        amount = graph[sender][receiver]["amount"]
+        print(
+            f"Potential Fraud Detected (Amount Exceeds $500.00): {sender} -> {receiver}, Amount: ${amount:.2f}"
+        )
     #   return the list of fraud edges that were found to caller
     return fraud_edges
 
 
+#
+# #####
+# the calculate_degree_centrality function
+#   input : graph - this is the networkx object that contains all
+#                   the nodes and edges from our csv file
+#           N - the number of top users with the highest degree of centrality to be printed
+#   output : this function does not return any value but prints the top N users with the highest degree of centrality
+#   Function explanation:
+#       We call nx.degree_centrality() to calculate the degree centrality of each node in the graph
+#       Then we sort the dictionary returned by nx.degree_centrality() in descending order of values
+#       Finally, we print the first N items from the sorted dictionary, which represent the top N users with the highest degree of centrality
+# #####
+def calculate_degree_centrality(graph, N):
```

Comment:
A lot of the changes are things my Python formatter does. The main thing is this function.